lib/portlayer/exec2/container.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exec2
+
+import (
+	"time"
+
+	"github.com/vmware/vic/pkg/vsphere/vm"
+)
+
+// All data in this struct must be data that is either immutable
+// or can be relied upon without having to query either the container guest
+// or the underlying infrastructure. Some of this state will be updated by events
+type container struct {
```

Comment:
How about arbitrary data fields?  Adding them may not be exposed to users, but it would be nice to have some sort of out of band data fields.  Kinda like docker's tags.  I am not saying we use the datafields for tags.  It can be used for other purposes.  Perhaps admin wants to add comments to a container.  Customer might want to add structure labeling.